q2_composition/assets/run_ancombc.R

```diff
@@ -69,20 +69,28 @@ output_loaf         <- opt$output_loaf
 if (!file.exists(inp_abundances_path)) {
   errQuit("Input file path does not exist.")
 } else {
-  otu_file <- t(read.delim(inp_abundances_path, check.names = FALSE,
-                            row.names = 1))
+  # Tidyverse
+  otu_file <- inp_abundances_path |>
+    # The sample IDs are in the first column, which we treat as characters.
+    # All other columns are feature frequencies as doubles
+    read_tsv(col_names = T, col_types = cols("character", .default = col_double())) |>
+    # The first column is called "...1" during import, then we convert it to row names.
+    column_to_rownames("...1") |>
+    t()
   }
 
 if (!file.exists(inp_metadata_path)) {
   errQuit("Metadata file path does not exist.")
 } else {
-  metadata_file <- read.delim(inp_metadata_path, check.names = FALSE,
-                              row.names = 1)
+  md_file <- read.delim(inp_metadata_path,
+                        check.names = FALSE, row.names = 1,
+                        colClasses = list(`sample-id` = "character"))
```

Comment:
Should we maybe use either tidyverse importing funcs or the base ones (but not both) for consistency? Then in the future a maintainer wouldn't need to research both `col_types` and `colClasses` for example.

Reply:
ah yeah that's probably a good call @colinvwood, thanks for pointing that out. i'll update md_file to use tidyverse